core/io/autoclose_spec.rb

```diff
@@ -0,0 +1,49 @@
+require_relative '../../spec_helper'
+require_relative 'fixtures/classes'
+
+describe "IO#autoclose" do
+  before :each do
+    @io = IOSpecs.io_fixture "lines.txt"
+  end
+
+  after :each do
+    @io.autoclose = true unless @io.closed?
+    @io.close unless @io.closed?
+  end
+
+  it "can be set to true" do
+    @io.autoclose = true
+    @io.should.autoclose?
+  end
+
+  it "can be set to false" do
+    @io.autoclose = false
+    @io.should_not.autoclose?
+  end
+
+  it "can be set to any truthy value" do
+    @io.autoclose = 42
+    @io.should.autoclose?
+  end
+
+  it "can be set multple times" do
+    @io.autoclose = true
+    @io.should.autoclose?
+
+    @io.autoclose = false
+    @io.should_not.autoclose?
+
+    @io.autoclose = true
+    @io.should.autoclose?
+  end
+
+  it "cannot be queried on a closed IO object" do
+    @io.close
+    -> { @io.autoclose? }.should raise_error(IOError, /closed stream/)
+  end
+
+  it "cannot be set on a closed IO object" do
+    @io.close
+    -> { @io.autoclose = false }.should raise_error(IOError, /closed stream/)
+  end
+end
```

Comment:
Looks good, but it makes sense to have separate test cases for the methods `#autoclose=` and `#autoclose?`.

There could also be added the following checks:
- `nil` value is treated as `false` in `#autoclose=`
- by default `#autoclose?` returns `true` (it's actually describe behaviour of a constructor method)